open-metadata-resources/open-metadata-deployment/charts/odpi-egeria-vdc/templates/apache-ranger-admin.yaml

```diff
@@ -74,7 +74,7 @@ spec:
           image: "{{ if (.Values.image.configure.registry | default .Values.imageDefaults.registry) }}{{ .Values.image.configure.registry | default .Values.imageDefaults.registry }}/{{ end }}\
                   {{ if (.Values.image.configure.namespace | default .Values.imageDefaults.namespace) }}{{ .Values.image.configure.namespace | default .Values.imageDefaults.namespace }}/{{ end }}\
                   {{ .Values.image.configure.name }}\
-                  :{{ .Values.image.configure.tag | default .Values.imageDefaults.tag }}"
+                  :{{ .Values.image.configure.tag | default .Values.egeria.version }}"
```

Comment:
One example (others in solr, egeria, ibm-igc, and the init jobs) -- for consistency this needs to be `.Values.image.egeria.tag` rather than `.Values.image.configure.tag` (to pickup the singular Egeria version rather than a configure-specific tag); though see previous comments about whether this condition is not backwards at the moment...

Reply:
Actually this was my intent - to allow the configure image tag to be overriden if required, but generally to default to the release version, as encoded by default in values.yaml.

So normally the 1.5-SNAPSHOT chart will be using the 1.5-SNAPSHOT docker image (which itself is made available very soon after merge - if the PR works!)

So it's not consistent with ranger, atlas - but intentionally so. The versioning of those images may need further work, but there's not an immediate pressure as we aren't really revisiting that scenario until 1.8 - of course that may change, but if so will revisit them then.

For now the old versioning and build process remains in place... and when we change it will be more radical than just a version change, I think we'll pull the build etc completely out into a seperarate third party repo.

Reply:
I'm not sure I follow...  This would mean that within the same chart we'll end up with different versions of the `configure` image being deployed into different pods, which I think is confusing (and potentially error-prone, ie. if we delete a particular version or fix a bug in a particular version but there is then a mix of versions deployed)?  These aren't the versions of the ranger, atlas, solr etc images themselves -- just the configure image that contains various utilities (curl, etc) used for running setup / initialization.

I would suggest we keep the same version of this configure image across everything.  This would also mean that as we upgrade that image with new base images (alpine) or other fixes to eg. vulnerabilities that we gain those fixes across all places `configure` is used rather than just the core Egeria ones.

(Even if we break these other images out into their own repo, they'd still need such a `configure` image to do their configuration, which I'd suggest re-using from Egeria core rather than creating a side version of.)

Reply:
I agree we wouldn't want mixed versions of a specific image in a deployment

I've gone through all the yaml's again.

Each time we use the egeria image I use the form
```
:{{ .Values.image.egeria.tag | default .Values.egeria.version }}
```
Each time I use the configure image I use:
```
:{{ .Values.image.configure.tag | default .Values.egeria.version }}
```

My build changes have changed/improved the build of these two images - configure & egeria. They are built after merge. Both are built with the version of egeria they are built with (this can be overriden, they are also tagged as latest for master only)

The default values.yaml sets 'egeria.version=1.5-SNAPSHOT', which is aligned with the release currently in master. 

Thus we will use this version for both egeria and configure across both charts & all jobs/pods.

The user may also set either, or both of
 image.configure.tag
 image.egeria.tag

If set these override the 'inherited' egeria.version set above (I name it like this to be very clear it's a common setting not image specific). They can be different, but each time configure is used it will always be the same. Ditto for egeria.

There's no change to the way our other images are used (though I do plan to change notebooks in the near future)

Did I miss an update?
Have I interpreted the golang expression wrongly?

Reply:
> Each time I use the configure image I use:  :{{ .Values.image.configure.tag | default .Values.egeria.version }}

I think I misinterpreted which was correct, then -- there are a few places where you're setting up the configure image differently than this (see example in comment below on the apache-atlas template)

Reply:
I think it was just one? Should now be corrected.

Reply:
I went back to double-check and couldn't find another, so I think you're right!